### DISCUSSION TITLE
GetFirstBrokenRule when we have an object graph

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3355] [Questions]
We have a parent object with several 'child' objects. We would like to show the 'first' broken rules for the whole object.
For single objects, it was easy we used 'GetRegisteredProperties()' to get the list of all properties and then iterate the broken rules collections, essentially our code looked like this
 ```cs
       foreach (var propertyinfo in 
            Csla.Core.FieldManager.PropertyInfoManager.GetRegisteredProperties(typeof(Employee)))
            {
                var rule = objEmployee.BrokenRulesCollection.GetFirstBrokenRule(propertyinfo);
                if (!(rule is null))
                {
                    if (rule.Severity == RuleSeverity.Error)
                        message += "- " + rule.Description + Environment.NewLine;

                }

            }

```

In an object graph, we get the list of 'BrokenRulesNode' objects. But each BrokenRulesNode object has a generic 'Object' for the business object associated with the node, making it tough to get a list of all properties.

I wonder if there is a way to get the First broken rules from 'BrokenRules' without iterating through the list of properties.

Any ideas?

Kind Regards

**Comment by rockfordlhotka** [comment]
What do you mean by "first"? First rule for each object, or for each property, or for the whole graph?

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  first rule for each property in the whole object graph.
